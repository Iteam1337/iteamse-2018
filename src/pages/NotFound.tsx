import * as React from 'react'
import { Helmet } from 'react-helmet'
import { GridColumnClean } from '../components/Grid/GridColumn'
import HeaderClean from '../components/Header/HeaderClean'
import Link from '../components/Link/Link'
import H1 from '../components/Typography/H1'
import Paragraph from '../components/Typography/Paragraph'
import styled from '../theme'
import brokenComputer from './img/illustrations-mac.png'

const Wrap = GridColumnClean.extend`
  display: grid;
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;

  grid-template-columns: 30px 1fr 30px;
  grid-template-rows: auto 1fr;
  background-color: #F3F2F2;
  overflow-y: scroll;
`

const Content = styled.div`
  display: grid;
  grid-row: 2;
  grid-column: 2;
  grid-column-gap: 50px;
  grid-template-rows: 200px 1fr;

  @media (min-width: 480px) {
    margin-top: 150px;
    grid-template-rows: auto 1fr;
    grid-template-columns: 1fr 2fr;
  }
`

const Broken = styled.img`
  height: 100%;
  @media (min-width: 480px) {
    height: auto;
  }
`

const StyledH1 = H1.extend`
  margin-bottom: 15px;
`

const Message = styled.div`
  margin-top: 20px;
  @media (min-width: 480px) {
    margin-top: 0;
    align-self: center;
  }
`
const NarrowLink = Link.extend`
  display: inline-block;
  @media only screen and (device-width: 375px) and (device-height: 812px) and (-webkit-device-pixel-ratio: 3) {
    letter-spacing: -0.5px;
  }
`

export class NotFound extends React.Component {
  render() {
    const message =  `Det blir inte alltid som man tänkt sig.
    Tyvärr finns inte sidan du försökte nå, antingen är länken fel eller så har sidan flyttats.
    Men du behöver inte oroa dig, vi ska göra allt vi kan för att detta inte ska hända igen.`
    return (
      <>
        <Wrap>
          <Helmet>
            <title>Iteam | Page not found</title>
          </Helmet>
          <HeaderClean />
          <Content>
            <Broken alt="broken" src={brokenComputer} />
            <Message>
              <StyledH1>
                Oj, någonting gick snett!
              </StyledH1>
              <Paragraph>{message}
              </Paragraph>
              <NarrowLink to="/">
                Ta mig tillbaka till startsidan
              </NarrowLink>
            </Message>
          </Content>
        </Wrap>
      </>
    )
  }
}

export default NotFound
