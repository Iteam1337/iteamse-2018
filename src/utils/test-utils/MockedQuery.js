import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import ApolloClient from 'apollo-client'
import { createHttpLink } from 'apollo-link-http'
import createMockedNetworkFetch from 'apollo-mocknetworkinterface'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'

type P = {
  children: React.Node,
  loading?: boolean,
  response: Object,
}

const MockedQuery = ({ children, loading, response = {} }: P) => {
  const createResponse = () => {
    return {
      loading: !!loading,
      data: {
        ...response,
      },
    }
  }

  const mockedNetworkFetch = createMockedNetworkFetch(createResponse, {
    latency: 0,
  })

  const client = new ApolloClient({
    link: createHttpLink({
      uri: 'http://localhost:3000',
      fetch: mockedNetworkFetch,
    }),
    cache: new InMemoryCache(),
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <BrowserRouter>{children}</BrowserRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedQuery
