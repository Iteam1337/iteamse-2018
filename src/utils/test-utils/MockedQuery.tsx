import { InMemoryCache } from 'apollo-cache-inmemory'
import ApolloClient from 'apollo-client'
import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import { mockSingleLink } from 'react-apollo/test-utils'
import MockRouter from 'react-mock-router'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'

interface MockedQueryProps {
  location?: object
  mocks: any[]
  params?: object
}

const MockedQuery: React.SFC<MockedQueryProps> = ({
  children,
  location,
  mocks,
  params,
}) => {
  const link = mockSingleLink.apply(null, mocks)

  const client = new ApolloClient({
    cache: new InMemoryCache(),
    link,
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <MockRouter location={location} params={params}>
          {children}
        </MockRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedQuery
