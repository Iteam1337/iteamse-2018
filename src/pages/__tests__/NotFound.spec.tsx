import React from 'react'
import { render, wait } from 'react-testing-library'
import MockedQuery from '../../utils/test-utils/MockedQuery'
import { NotFound } from '../NotFound'

describe('components/NotFound', () => {
  it('renders NotFound', async () => {
    const { container } = render(
      <MockedQuery mocks={[]}>
        <NotFound />
      </MockedQuery>
    )

    await wait()

    expect(container).toMatchSnapshot()
  })
})
