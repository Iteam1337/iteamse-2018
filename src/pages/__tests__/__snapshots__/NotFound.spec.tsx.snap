// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NotFound renders NotFound 1`] = `
<div>
  <div
    class="sc-iwsKbI diJSmU"
  >
    <div
      class="sc-bxivhb fsBQqP"
    >
      <div
        class="sc-bZQynM znEUq"
      >
        <div
          class="sc-ifAKCX cKzunR"
        >
          <a
            class="sc-htpNat frOpea"
          >
            <img
              class="sc-EHOje jaKvOm"
              alt="Logo"
              src="test-file-stub"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="sc-gZMcBi ilQTGq"
    >
      <img
        class="sc-gqjmRU hwXSZV"
        alt="broken"
        src="test-file-stub"
      />
      <div
        class="sc-jTzLTM eWcbFE"
      >
        <h1
          class="sc-VigVT btjUSt"
        >
          Oj, någonting gick snett!
        </h1>
        <div
          class="sc-dnqmqq oKUkC"
        >
          <div>
            <p>
              404 error. Eller som man säger på svenska, något har blivit fel. Tyvärr finns inte sidan du försökte nå, antingen är länken fel eller så har sidan flyttats. Men du behöver inte oroa dig, vi ska göra allt vi kan för att detta inte ska hända igen.
            </p>
          </div>
        </div>
        <a
          class="sc-fjdhpX iJPrzS"
        >
          Ta mig till startsidan
        </a>
      </div>
    </div>
  </div>
</div>
`;
